### ISSUE TITLE
RKE2 multi-node Cluster unable to scale down correctly

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#32838]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
* deploy a multi-node downstream cluster (say, with 2 extra worker nodes)
* allow cluster to complete provisioning
* edit cluster, and update the machine count to 1

**Result:**
* nodes stuck in `deleting` state through rancher UI, but still shows nodes as available
<img width="154" alt="Screen Shot 2021-05-21 at 3 55 55 PM" src="https://user-images.githubusercontent.com/16691014/119205509-405cbe00-ba4d-11eb-8e45-8099a34f064b.png">

**Other details that may be helpful:**
* the nodes are actually removed from the provisioner, rancher just isn't removing them correctly

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): master-head (4f9850d)
- Installation option (single install/HA): single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Digital Ocean
- Machine type (cloud/VM/metal) and specifications (CPU/memory): 2vCPU, multiple nodes
- Kubernetes version (use `kubectl version`):v1.21.1-alpha5+rke2r1

**Comment by deniseschannon** [comment]
As part of removing/scaling down nodes, CAPI, which is what the new provisioning framework is based on automatically drains nodes, which might be part of the issue.

@slickwarren How are you removing nodes in this use?

Are you deleting it from the UI? Are you sizing down the node pool?

**Comment by slickwarren** [comment]
I am deleting it through the UI using `edit Config -> machine count`
<img width="226" alt="Screen Shot 2021-06-03 at 2 09 33 PM" src="https://user-images.githubusercontent.com/16691014/120712347-5d45b800-c475-11eb-8a82-f670d41c0d2e.png">

**Comment by deniseschannon** [comment]
@slickwarren I updated the comment to reflect your note. 

By default, CAPI automatically drains when removing nodes, but since drain often has issues, we will look into changing the behavior around drain so that it's off by default and a user can opt in.

**Comment by deniseschannon** [comment]
Currently blocked by https://github.com/rancher/dashboard/

Unable to edit any config of any cluster, but scaling down using the `-` button on the machine pools page works.

**Comment by deniseschannon** [comment]
Note: I think this works (outside of my broken setup), but it shoudl be validated and there is a separate issue around the nodes in the provider are now not being cleaned up.

**Comment by thedadams** [comment]
This seems to be working now. I was able to scale worker nodes up and down.

**Comment by slickwarren** [comment]
I was unable to reproduce the issue on 2.6-head (4def656)
